### PR TITLE
Apply misc build, CI and testing improvements

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -81,12 +81,6 @@ class ResolveIntelliJVersions extends DefaultTask {
 task resolveIntelliJVersions(type: ResolveIntelliJVersions)
 
 
-if (isCI) {
-  gradle.startParameter.configureOnDemand = false
```
Comment (PawelLipski) [inline build.gradle:85]: Not sure why that was even needed :thinking:

File: build.gradle [hunk 2]
```diff
@@ -81,12 +81,6 @@ class ResolveIntelliJVersions extends DefaultTask {
...
-  gradle.startParameter.logLevel = LogLevel.INFO
-  gradle.startParameter.showStacktrace = ShowStacktrace.ALWAYS
```
Comment (PawelLipski) [inline build.gradle:87]: These should (hopefully) be now addressed with `testLogging` settings instead, leading to less spam from Gradle, only the failed test output that we really care about and that's not show in Gradle output by default

File: scripts/run-pre-build-checks [hunk 3]
```diff
@@ -10,7 +10,6 @@ export GIT_PAGER=cat
 ./scripts/enforce-indent-two-spaces-outside-java
 ./scripts/enforce-issue-number-for-todos
 ./scripts/enforce-newline-at-eof
-./scripts/enforce-onupdate-call-super-if-overridden
```
Comment (PawelLipski) [inline scripts/run-pre-build-checks:13]: Less hacks, more ArchUnit

File: scripts/run-ui-tests [hunk 4]
```diff
@@ -26,6 +26,5 @@ only_tests_flag=${test_method_prefix:+--tests *.$test_method_prefix*}
 for version in $versions; do
   ./gradlew \
     -PintellijVersionForUiTests="$version" \
-    --info \
```
Comment (PawelLipski) [inline scripts/run-ui-tests:29]: Again, controlling the output here using `testLogging`

File: src/test/java/com/virtuslab/archunit/ClassStructureTestSuite.java [hunk 5]
```diff
@@ -18,6 +22,37 @@ public void actions_implementing_DumbAware_should_extend_DumbAwareAction() {
         .check(importedClasses);
   }
 
+  @Test
+  public void actions_overriding_onUpdate_should_call_super_onUpdate() {
+    classes()
+        .that()
+        .areAssignableTo(com.virtuslab.gitmachete.frontend.actions.base.BaseProjectDependentAction.class)
+        .and()
+        .areNotAssignableFrom(com.virtuslab.gitmachete.frontend.actions.base.BaseProjectDependentAction.class)
```
Comment (PawelLipski) [inline src/test/java/com/virtuslab/archunit/ClassStructureTestSuite.java:31]: Basically, **subclasses** of `BaseProjectDependentAction`... I can't see any easier way to express that, only with these upper/lower bounds criteria

File: src/test/java/com/virtuslab/archunit/ClassStructureTestSuite.java [hunk 6]
```diff
@@ -18,6 +22,37 @@ public void actions_implementing_DumbAware_should_extend_DumbAwareAction() {
...
+        .and(new DescribedPredicate<JavaClass>("override onUpdate method") {
+          @Override
+          public boolean apply(JavaClass input) {
+            return input.getMethods().stream().anyMatch(method -> method.getName().equals("onUpdate"));
```
Comment (PawelLipski) [inline src/test/java/com/virtuslab/archunit/ClassStructureTestSuite.java:35]: `get(Declared)Methods`, as opposed to `getAllMethods`, which would include the inherited ones as well

File: .circleci/config.yml [hunk 7]
```diff
@@ -29,22 +44,18 @@ jobs:
...
-          name: Start Gradle daemon
-          command: ./gradlew
+          name: Validate Markdown values
+          # Currently, just check whether earliest supported IntelliJ version in README.md
+          # is up to date with the source of truth (Gradle config).
+          command: |
+            expected=$(./gradlew -q resolveIntelliJVersions --versionKey=earliestSupportedMajor)
+            actual=$(grep -Po '(?<=minimum required version is )[0-9.]+' README.md)
+            if [[ $actual != "$expected" ]]; then
+              echo "README.md indicates that the minimum required IntelliJ version is $actual, while in fact in Gradle config it is $expected"
```
Comment (mkondratek) [inline .circleci/config.yml:55]: :rocket:

**Review by mkondratek — APPROVED** [review]

**Review by micpiotrowski — APPROVED** [review]

**Pull request merged**